linode/manager
# feat: [M3-7284, M3-7361] - Create VPC drawer in Linode Create flow; Update widths for Create VPC and Create Firewall buttons

## Description 📝
- Add Create VPC drawer to the Linode Create flow. Now, if a user wants to create a VPC while creating a linode, they can do so while staying on the same page. No more new tabs!
- Update 11/6: including M3-7361 in this ticket (as it'd been caught and handled here earlier anyway)

## Changes  🔄
List any change relevant to the reviewer.
- Add `CreateVPCDrawer`
- Abstracted a lot of common logic between `VPCCreate` page and `VPCCreateDrawer`, creating a new custom hook and other components
- Added some tests for new components
- Created new constants in the VPC's constants file for some text 
- some VPC Create page cosmetic changes - Right aligned 'Create VPC' button on VPC create page, removed divider between last subnet and 'Add another subnet' button

## Preview 📷
**Include a screenshot or screen recording of the change**
(the grey shadow in the second img is from my computer's menu bar)

Region that doesn't support vpcs selected:
![image](https://github.com/linode/manager/assets/139280159/503c1724-68f2-44fa-89d9-f676ab9c9690)


| drawer img 1  | drawer img 2   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/8af9f2d6-5dad-4893-9929-f58a64ff3c06) | ![image](https://github.com/linode/manager/assets/139280159/559a370b-72c9-4b32-bea9-1b208f0244f5) |

## How to test 🧪

### Prerequisites
- Using the dev environment, with vpc admin tags

### Verification steps 

On the Linode Create page
- Verify that if you do not have a region selected, or have a [region that does not support VPCs selected](https://github.com/linode/manager/pull/9787), the Create VPC does not appear and there's instead text that says "VPC is not available in the selected region."
- If the Create VPC link shows up:
  - Verify that the region you've chosen is pre-selected and the region drop down is disabled
  - Verify VPC Create functionality (it should work the same as the VPC Create page EXCEPT that you must have at least 1 subnet in this flow)
  - Verify that after creating a VPC, the VPC is selected in the VPC panel and that creating a linode with that VPC works as expected

On the VPC Create page
- verify the 'Create VPC' button is now right aligned, not left aligned
- verify there's no divider between the last subnet and the 'Add another subnet' button (and that there's increased spacing between the last subnet and add subnet button)
- verify that besides those changes ^, the VPC Create page looks the same
- verify that the VPC Create page still functions the same 

On the VPC details page
- verify the Create Subnet drawer still looks the same (and still functions the same, although its functionality shouldn't have been affected by these changes)

### Tests
- `yarn test VPC` to run all the VPC related tests
- `yarn cy:run -s 'cypress/e2e/core/vpc/vpc-create.spec.ts`  to make sure there are no regressions here

## As an Author I have considered 🤔

*Check all that apply*
